libgalois/src/analytics/louvain_clustering/louvain_clustering.cpp

```diff
@@ -382,18 +383,46 @@ struct LouvainClusteringImplementation
     }
     std::vector<std::string> temp_edge_property_names = {
         "_katana_temporary_property_" + edge_weight_property_name};
+
+    temp_edge_property_names.push_back(
+        "_katana_temporary_property_" + edge_type_property_name);
     if (auto result = ConstructEdgeProperties<EdgeData>(
             pfg_mutable.get(), temp_edge_property_names);
         !result) {
       return result.error();
     }
 
+    //add here
     auto graph_result = Graph::Make(pfg);
     if (!graph_result) {
       return graph_result.error();
     }
     Graph graph_curr = graph_result.value();
 
+    if (!edge_type_property_name.empty()) {
+      katana::do_all(
+          katana::iterate((uint64_t)0, graph_curr.num_nodes()),
+          [&](uint64_t n) {
+            graph_curr.template GetData<CurrentCommunityId>(n) = n;
```

Comment:
This should be done after edge projection

Reply:
I am using this for graph filtering. They are re-initialied anyways at the start of a  new coarsened graph iteration. thanks

Reply:
Why do you need this for filtering?

Reply:
got it now, thanks